pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,107 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  nodejs,
+  pnpm_9,
+  # cpp is required for building workerd.
+  # workerd is used for some wrangler subcommands.
+  # non-linux platforms may experience errors
+  # on certain subcommands due to this issue.
+  autoPatchelfHook,
+  llvmPackages,
+  musl,
+  xorg,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  };
+
+  buildInputs =
+    # cpp is required for building workerd.
+    # workerd is used for some wrangler subcommands.
+    # non-linux platforms may experience errors
+    # on certain subcommands due to this issue.
+    lib.optional stdenv.isLinux
```

Comment:
I think it'd be more correct to add these inputs unconditionally and use the `meta.platforms` attribute.

Reply:
ok i can do that, i just didn't want to make wrangler limited on mac, afaik they normally can download and use wrangler just fine, but the normal install is just pulling the binary from npm or something. i figured it would be better to at least let them use the commands that do work but i can see either way.

Reply:
> ok i can do that, i just didn't want to make wrangler limited on mac, afaik they normally can download and use wrangler just fine, but the normal install is just pulling the binary from npm or something. i figured it would be better to at least let them use the commands that do work but i can see either way.

Yes but it would be too hard to maintain that, especially if you don't have a Darwin machine to test it locally. When a darwin user will want to use this package, they'll make the effort don't worry.